tests/test-class-amp-theme-support.php

```diff
@@ -330,4 +330,25 @@ public function test_handle_xhr_request() {
 		AMP_Theme_Support::handle_xhr_request();
 		$this->assertContains( 'AMP-Access-Control-Allow-Source-Origin: https://example.org', xdebug_get_headers() );
 	}
+
+	/**
+	 * Test ensure_required_markup().
+	 *
+	 * @covers AMP_Theme_Support::ensure_required_markup()
+	 */
+	public function test_ensure_required_markup() {
+		$schema_test_value = '<script type="application/ld+json">{"@context":"http:\/\/schema.org"';
+
+		$page = '<html><head></head><body>Test</body></html>';
+		$dom  = new DOMDocument();
+		$dom->loadHTML( $page );
+		AMP_Theme_Support::ensure_required_markup( $dom );
+		$this->assertContains( $schema_test_value, $dom->saveHTML() );
+
+		$page = '<html><head<script type="application/ld+json">{"@context":"http:\/\/schema.org","publisher":{"@type":"Organization","name":"Test Blog"}}</script>></head><body>Test</body></html>';
+		$dom  = new DOMDocument();
+		$dom->loadHTML( $page );
+		AMP_Theme_Support::ensure_required_markup( $dom );
+		$this->assertEquals( substr_count( $dom->saveHTML(), $schema_test_value ), 1 );
```

Comment:
Also should add a test for when JSON is output but the slashes are not escaped.

Reply:
And add a test for when `@context` is not the first key.

Reply:
I made tests for when @anothercontext is the first key and has schema.org present, I hope i understood correctly.